### DISCUSSION TITLE
Separate frontend and backend

### DISCUSSION THREAD

**Discussion by robdejonge** [Koenkk/zigbee2mqtt#13721] [General]
The fact that we can add `frontend: false` to our configuration file (also the default, I believe), obviously means that we can run only the backend of the application somewhere. I was wondering if this means the frontend can be run in isolation too. 

I run z2m on a Pi Zero that is positioned for optimal signal, but would run the frontend on the virtual machine I have Home Assistant running on. I guess the frontend does not use a whole lot of resources, but still ...

**Comment by mausch** [comment]
You can, see https://www.npmjs.com/package/zigbee2mqtt-frontend

**Reply by robdejonge** [reply]
Interesting. A few questions I can't see answered on that site or the related repo: 
1) Can this run anywhere, or does it need to be on the same machine as z2m?
2) To run this, do I need to enable `frontend: true` on the z2m install?

Thanks for the suggestion!

**Reply by mausch** [reply]
1. I've not tried it but AFAICT it can run anywhere, just set the Z2M_API_URI env var
2. Pretty sure [you still need it enabled, yes, for the websocket server](https://github.com/Koenkk/zigbee2mqtt/blob/a9b8808237554316c5c97e9d9c990048733779e8/lib/controller.ts#L82).

like koenkk said you don't really save much by putting the frontend somewhere else, I think you only save the static file serving really.

**Reply by robdejonge** [reply]
Thanks, much appreciated!

**Comment by Koenkk** [comment]
The frontend runs in your browser, therefore the only difference between `frontend: true`/`frontend: false` is that with `true` a websocket server is started which is used by the frontend to communicate with z2m. So IMO there would be no benefit of this.

**Reply by robdejonge** [reply]
Thanks for the explanation. Much appreciated.
